### PR TITLE
CPP: Query tags 2 (JSF queries)

### PR DESCRIPTION
**Pull request by geoffw0**
Tag all of the JSF queries.  In a number of cases I wasn't sure whether to go for `maintainability` or `correctness`, but I tried to base my choice of tags on the stated intentions from the JSF doc.  In many cases I've also reduced the `@problem.severity` from `error` to `warning` or `recommendation`.  None of these changes should effect which queries are run on LGTM.

### CODE REVIEW

File: cpp/ql/src/jsf/4.21 Operators/AV Rule 165.ql [hunk 1]
```diff
@@ -4,13 +4,22 @@
...
+where ume.getOperand().getFullyConverted().getUnderlyingType().(IntegralType).isUnsigned()
       and not ume.getOperand() instanceof Literal
-select ume, "The unary minus operator should not be applied to an unsigned expression."
+//select ume, "The unary minus operator should not be applied to an unsigned expression."
+select
+	ume,
+	ume.getOperand().getUnderlyingType(),
+	ume.getOperand().getFullyConverted().getUnderlyingType(),
+	ume.getUnderlyingType(),
+	ume.getFullyConverted().getUnderlyingType()
```
Comment (jbj) [inline cpp/ql/src/jsf/4.21 Operators/AV Rule 165.ql]: It appears you left some test code behind here. Fortunately it was caught by PR-Checks.

File: cpp/ql/src/jsf/4.09 Style/AV Rule 41.ql [hunk 2]
```diff
@@ -3,7 +3,10 @@
  * @description Source lines will be kept to a length of 120 characters or less.
  * @kind problem
  * @id cpp/jsf/av-rule-41
- * @problem.severity warning
+ * @problem.severity warrecommendationning
```
Comment (jbj) [inline cpp/ql/src/jsf/4.09 Style/AV Rule 41.ql]: Typo: "warrecommendationning"

Comment (geoffw0) [reply cpp/ql/src/jsf/4.09 Style/AV Rule 41.ql]: Fixed.  Thanks for spotting that.

**Comment by geoffw0** [comment]
Rebased, fixed merge, removed some debug code that had snuck in.

**Pull request merged**